### PR TITLE
jewel: mon: crash on shutdown, lease_ack_timeout event

### PR DESCRIPTION
**Pull request by asheplyakov**
http://tracker.ceph.com/issues/19926

### CODE REVIEW

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -4741,7 +4757,9 @@ void Monitor::scrub_event_start()
     return;
   }
 
-  scrub_event = new C_Scrub(this);
+  scrub_event = new C_MonContext(this, [this](int) {
```
Comment (tchaikov) [inline src/mon/Monitor.cc]: might want to remove `C_Scrub` and `C_ScrubTimeout` in the header? i.e. we might need to backport #10513 also?

**Comment by asheplyakov** [comment]
@tchaikov I've cherry-picked e58750127cbe8728f4047411904af3890966490f and a few related commits (which move the code from header files to *.cc ones)

**Comment by tchaikov** [comment]
@asheplyakov much better, thanks!

**Comment by smithfarm** [comment]
Jenkins re-test this please

**Comment by smithfarm** [comment]
@tchaikov @jdurgin This passed a rados suite with 1 failure ( a segfault - see http://tracker.ceph.com/issues/21063#note-2 ). Do you think it can be merged?

**Comment by tchaikov** [comment]
@smithfarm i think this can be merged.

**Pull request merged**